python/ray/ml/predictors/integrations/rl/rl_predictor.py

```diff
@@ -0,0 +1,117 @@
+import os
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+import ray.cloudpickle as cpickle
+from ray.ml import Preprocessor, Checkpoint
+from ray.ml.constants import PREPROCESSOR_KEY
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.train.integrations.rl.rl_trainer import (
+    RL_TRAINER_CLASS_FILE,
+    RL_CONFIG_FILE,
+)
+from ray.rllib.agents.trainer import Trainer as RLlibTrainer
+from ray.rllib.utils.typing import EnvType
+
+
+class RLPredictor(Predictor):
+    """A predictor for RLlib trainer checkpoints.
+
+    Args:
+        trainer: The RLlib trainer instance containing the policy on which to
+            perform inference on.
+        preprocessor: A preprocessor used to transform data batches prior
+            to prediction.
+    """
+
+    def __init__(
+        self,
+        trainer: RLlibTrainer,
+        preprocessor: Optional[Preprocessor] = None,
+    ):
+        self.trainer = trainer
```

Comment:
I'm assuming in the future, we would be able to isolate single policies/models from a e.g. multi-agent checkpoint?

Reply:
Yeah I'm thinking about just passing a `policy` here instead of a trainer. What do you think - will this be more future proof? Is there any reason why we would need full access to the whole trainer and not just the policy?

Reply:
+1. Would checkpointing just the policies be sufficient? It seems a bit weird needing the entire trainer for prediction.